include/zephyr/device.h

```diff
@@ -1058,8 +1058,47 @@ __syscall int device_init(const struct device *dev);
 
 DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_DEVICE_DECLARE_INTERNAL)
 
+/** @brief Expands to the full type. */
+#define Z_DEVICE_API_TYPE(_class) _CONCAT(_class, _driver_api)
```

Comment:
Please avoid the _TYPE_ suffix if anyhow possible. Type is so overloaded with meaning...
```suggestion
#define Z_DRIVER_API_SECTION_NAME(_driver_api_name) _CONCAT(_driver_api_name, _driver_api)
```